src/run.ts

```diff
@@ -25,6 +40,12 @@ export async function run(context: ContextPlugin) {
     return result.logMessage.raw;
   }
 
+  if (!(await isUserAllowedToGeneratePermits(context))) {
+    const result = logger.error("You are not allowed to generate permits.");
+    await postComment(context, result);
+    return result.logMessage.raw;
+  }
```

Comment:
isn't there supposed to be a config to enable/disable permit generation by outside collaborators?
also why not just throw the error and let sdk handle it?

Reply:
I did not find such setting in https://github.com/ubiquity-os-marketplace/text-conversation-rewards/blob/8fda635ef34b8659955b1dc993ff970ee40bff85/src/types/plugin-input.ts so I don't think so?

Yes that would be possible but the post on error was disabled on this plugin due to the complexity of messages and reuse of the previous message to update it. I can take care of that in a separate PR.

Reply:
I guess we could add the setting but I feel like it's not a priority since that's such a rare use case because why would anyone want someone outside the org to generate permits

Reply:
We needed it in the past for special purposes. Basically it was to work for a "self serve" demo.

Reply:
Should I add this or is it not needed?

Reply:
You should retain the capability and leave it to be configured, with its default to only allow collaborators to generate permits.

Reply:
Added option to enable / disable: https://github.com/ubiquity-os-marketplace/text-conversation-rewards/pull/209/commits/e73c062b689187cc8047accc1deb6bf89d7c0d3e
Example: https://github.com/Meniole/text-conversation-rewards/issues/41#issuecomment-2538271098